src/gui/sharelinkwidget.cpp

```diff
@@ -234,7 +236,9 @@ void ShareLinkWidget::slotSharesFetched(const QList<QSharedPointer<Share>> &shar
             _ui->widget_shareLink->show();
             _ui->checkBox_shareLink->setChecked(true);
 
-            _ui->checkBox_password->setEnabled(true);
+            if (!_passwordRequired) {
+                _ui->checkBox_password->setEnabled(true);
+            }
```

Comment:
how about doing _ui->checkBox_password->setEnabled( !_passwordRequireed) ? That way it's shorter and you always have a defined state.